xmtp_mls/src/groups/mod.rs

```diff
@@ -326,8 +331,12 @@ impl MlsGroup {
         let welcome = deserialize_welcome(&welcome_bytes)?;
 
         let join_config = build_group_join_config();
-        let staged_welcome =
-            StagedWelcome::new_from_welcome(provider, &join_config, welcome.clone(), None)?;
+        let processed_welcome =
+            ProcessedWelcome::new_from_welcome(provider, &join_config, welcome.clone())?;
+        if processed_welcome.psks().is_empty() {
```

Comment:
@franziskuskiefer I've got some test failures because this `is_empty()` call runs forever and never completes. Any ideas of why that would happen?

Reply:
@nplasterer we had a look at this and noticed that the functions returns the `NoPSKSupport` error if the Welcome does not contain PSKs. After negating the condition of the `if` on like 336 all tests pass and we don't see any hangs. Does this work for you as well?

Reply:
Of course. This call is supposed to be !empty not isEmpty. Blame that on Friday brain. 🤦‍♀️ Appreciate y'all taking the time. 🙏